runtime/compiler/runtime/JITaaSIProfiler.cpp

```diff
@@ -514,9 +515,11 @@ TR_JITaaSClientIProfiler::serializeAndSendIProfileInfoForMethod(TR_OpaqueMethodB
       // Walk all bytecodes and populate the sorted array of interesting PCs (pcEntries)
       // numEntries will indicate how many entries have been populated
       // These profiling entries have been 'locked' so we must remember to unlock them
-      bool abort = false;
       bytesFootprint = walkILTreeForIProfilingEntries(pcEntries, numEntries, &bci, method, BCvisit, abort, comp);
-      if (numEntries && !abort)
+      if (abort)
```

Comment:
Even though abort is set, some of the entries could have been locked, so we need to run the sequence that unlocks them below.